addons/analytic/models/analytic_plan.py

```diff
@@ -14,6 +14,8 @@ class AccountAnalyticPlan(models.Model):
     _parent_store = True
     _rec_name = 'complete_name'
     _order = 'sequence asc, id'
+    _check_company_auto = True
+    _check_company_domain = models.check_company_domain_parent_of
```

Comment:
You should update the `account.analytic.applicability` instead (you already updated the record rules)

There is no `company_id` on the plans (anymore since 16.0)